pilot/proxy/envoy/discovery.go

```diff
@@ -365,9 +365,14 @@ func (ds *DiscoveryService) ListAllEndpoints(request *restful.Request, response
 					if instance.AvailabilityZone != "" {
 						t = &tags{AZ: instance.AvailabilityZone}
 					}
+					hostPort := instance.Endpoint.Port
+					if port.AuthMigrationPort != nil &&
+						port.AuthMigrationPort.Active {
+						hostPort = port.AuthMigrationPort.Port
+					}
```

Comment:
why do you have to rewrite the discovery request? all you need is to supply a different set of endpoints for SDS requests.

Reply:
nm.. ignore my comment